### PR TITLE
HWP: Now in .300 Blackout

### PR DESCRIPTION
**Pull request by DukePaulAtreid3s**
#### Summary
Content "Adds the "HWP SOCOM Configuration", chambered in .300 Blackout "

#### Purpose of change
5.56 and .300 BLK are very similar calibers, to the point where they can be held in the same magazines. This, compounded with the adoption of the SIG Rattler for special forces use, makes it a prime candidate for an  HWP conversion. 

#### Describe the solution
Adds the aforementioned modification, as well as the ability for 5.56 HWP mags to hold .300 . The SOCOM configuration is PDW length and integrally suppressed, due to .300 Blackout round's superior ballistics and subsonic performance in short barrels compared to 5.56. 

#### Describe alternatives you've considered
Making it extra short, adding .22LR HWP instead

#### Testing
Screenshot 2024-11-11 181837

#### Additional context
The first HWP conversion I've wanted to do that wasn't just a meme.

### CODE REVIEW

**Comment by DukePaulAtreid3s** [comment]
@John-Candlebury I hope this is up to snuff

**Comment by John-Candlebury** [comment]
Is 300blk a commonplace round? Is it reasonable to have this vs a 7.76 conversion that makes the gun heavier overall. 

I don't care much about guns in general but do care about th HWP not getting diluted to serve a bunch of cartridges with almost no in game use.

**Comment by DukePaulAtreid3s** [comment]
It is commonplace among special forces, due to its superior ballistics in short barrels than 5.56. It also is used by many civilians, partly because any AR-15 platform rifle can be converted to fire it with a simple barrel swap.

File: data/json/items/gun/robofac_gun.json [hunk 1]
```diff
@@ -310,6 +310,35 @@
...
+    "mod_targets": [ "robofac_gun" ],
+    "name": { "str_sp": "HWP STEALTH configuration" },
+    "mode_modifier": [ [ "DEFAULT", "semi-auto", 1 ] ],
+    "install_time": "1 m",
+    "ammo_modifier": [ "300blk" ],
+    "min_skills": [ [ "weapon", 1 ] ],
+    "dispersion_modifier": 12,
+    "loudness_modifier": -75,
+    "overwrite_min_cycle_recoil": 1200,
+    "description": "A Hub 01 Hybrid Weapons Platform quick swap conversion kit.  This version has a long, thick, heavy barrel with \".300 BLK STEALTH\" stamped into the metal, and an integrated suppressor sitting flush with it. It feeds from 5.56 magazines."
```
Comment (John-Candlebury) [inline data/json/items/gun/robofac_gun.json:340]:
```suggestion
    "description": "A Hub 01 Hybrid Weapons Platform quick swap conversion kit.  This version has a long barrel with \".300 BLACKOUT\" stamped into the metal, and a large integrated suppressor. It feeds from 5.56 magazines."
```

**Review by John-Candlebury — APPROVED** [review]
Looks good to me beyond those suggestions. I'll leave it up for a few days in case contributors with more gun knowledge want to review it.

**Comment by Holli-Git** [comment]
A) Why is it heavier and with a longer barrel than the 556 version? The main difference would just be a larger bore. 
B) Is .300 BLK actually common among special forces? I'd like to have verification for that
C) Does the HUB or its mercenaries really need a .300 BLK gun? Yeah suppressors are cool but there's already a suppressed version in 9mm

File: data/json/items/gun/robofac_gun.json [hunk 2]
```diff
@@ -567,10 +596,10 @@
     "material": [ "plastic", "steel" ],
     "symbol": "#",
     "color": "light_gray",
-    "ammo_type": [ "223" ],
+    "ammo_type": [ "223", "300blk" ],
     "flags": [ "MAG_BULKY" ],
     "mag_jam_mult": 2,
-    "pocket_data": [ { "pocket_type": "MAGAZINE", "rigid": true, "ammo_restriction": { "223": 100 } } ]
+    "pocket_data": [ { "pocket_type": "MAGAZINE", "rigid": true, "ammo_restriction": { "223": 100, "300blk": 100 } } ]
```
Comment (Holli-Git) [inline data/json/items/gun/robofac_gun.json:602]: I am not confident that 300 BLK works in drums

**Review by Holli-Git — CHANGES_REQUESTED** [review]

File: data/json/items/gun/robofac_gun.json (hunk 2 above)
Comment (DukePaulAtreid3s) [reply data/json/items/gun/robofac_gun.json:602]: I found listings for combined .300/.223/5.56 drums

Comment (Holli-Git) [reply data/json/items/gun/robofac_gun.json:602]: Can you share them?

**Comment by DukePaulAtreid3s** [comment]
> than the 556 version? The main difference would just be a larger bore.
> B) Is .300 BLK actually common among special forces? I'd like to have verification for that
> C) Does the HUB or its mercenaries really need a .300 BLK gun? Yeah suppressors are cool but there's already a suppressed version in 9mm

A.) That is a mistake from earlier being corrected.
B.) https://sam.gov/opp/0497b64a90ca4963a86d86df826d9281/view .
- [ ] C) The Hub currently lacks a suppressed marksman rifle, and given the current, proven use of the .300 blackout cartridge in that role, it makes sense to include due to the ease of conversion from 5.56 as well as the carryover between equipment.

**Comment by Holli-Git** [comment]
B) That just says it's used. No indication that it is "common" as you said. 
C) It lacking in something doesn't make it a role that needs to be fulfilled. Also, it seems you want this to both be a PDW and a DMR? We're not lacking in the DMR side with .308, and we're not lacking in the PDW side with 5.7 and 9mm

**Comment by DukePaulAtreid3s** [comment]
> B) That just says it's used. No indication that it is "common" as you said. 
> 
> C) It lacking in something doesn't make it a role that needs to be fulfilled. Also, it seems you want this to both be a PDW and a DMR? We're not lacking in the DMR side with .308, and we're not lacking in the PDW side with 5.7 and 9mm 
> 
> 

B) I will concede that it may not be "commonly" used, but it is used regularly enough for a government contract to be issued for guns, and the ammunition was developed specifically for SOCOM use. There's also evidence of its use by the SEALS
C) Its (essentially) a select-fire VAL, so it could fulfill both roles to a point. It should be similarly sized to the PDW

File: data/json/items/gun/robofac_gun.json (hunk 2 above)
Comment (DukePaulAtreid3s) [reply data/json/items/gun/robofac_gun.json:602]: https://www.highbyoutdoors.com/ProMag-AR-15-Drum-Magazine-223-Rem556-NATO300-Blk--65rd--Black-Polymer_p_77984.html

https://texasshooterssupply.com/product-category/magdrums/magdrums-223_556x45_300aac/

Comment (Holli-Git) [reply data/json/items/gun/robofac_gun.json:602]: The listing for the D-mag is lying, look at questions
image
https://magpul.com/pmag-d-60-ar-m4-gen-m3.html?mp_global_color=118

For the promag, the actual seller doesn't list it using .300 BLK 
https://promagindustries.com/ar-15-5-56mm-65-rd-black-polymer-drum/

**Comment by Holli-Git** [comment]
> B) I will concede that it may not be "commonly" used, but it is used regularly enough for a government contract to be issued for guns, and the ammunition was developed specifically for SOCOM use. There's also evidence of its use by the SEALS C) Its (essentially) a select-fire VAL, so it could fulfill both roles to a point. It should be similarly sized to the PDW

B) That doesn't really mean anything. The point is to be commonplace *enough*. Your evidence is one contract and I believe that to be unsatisfactory. The SEALS using it doesn't mean much, they're not gonna make it rain .300
C) What are you basing this on? The fact that they're both rifles in subsonic calibers? And no, you can't make it "similarly sized to a PDW" as the HWP is not a PDW. You aren't going to magically make it smaller

**Comment by DukePaulAtreid3s** [comment]
C: Similary sized to the Hub PDWs, which all have 14 in barrels

File: data/json/items/gun/robofac_gun.json [hunk 3]
```diff
@@ -420,17 +449,17 @@
...
-    "pocket_data": [ { "pocket_type": "MAGAZINE", "rigid": true, "ammo_restriction": { "223": 40 } } ]
+    "pocket_data": [ { "pocket_type": "MAGAZINE", "rigid": true, "ammo_restriction": { "223": 40, "300blk": 40 } } ]
   },
   {
     "id": "xedra40",
     "type": "MAGAZINE",
     "name": { "str": "prototype 40-round 5.56 magazine" },
     "description": "A high-capacity 40-round box magazine.",
     "copy-from": "robofac40",
-    "pocket_data": [ { "pocket_type": "MAGAZINE", "rigid": true, "ammo_restriction": { "223": 40 } } ]
```
Comment (Holli-Git) [inline data/json/items/gun/robofac_gun.json:433]: This is not a HWP magazine

**Review by Holli-Git — CHANGES_REQUESTED** [review]

File: data/json/items/gun/robofac_gun.json [hunk 4]
```diff
@@ -458,8 +458,17 @@
     "type": "MAGAZINE",
     "name": { "str": "prototype 40-round 5.56 magazine" },
     "description": "A high-capacity 40-round box magazine.",
-    "copy-from": "robofac40",
-    "pocket_data": [ { "pocket_type": "MAGAZINE", "rigid": true, "ammo_restriction": { "223": 40, "300blk": 40 } } ]
+    "weight": "265 g",
```
Comment (Holli-Git) [inline data/json/items/gun/robofac_gun.json:461]: What was the purpose of this?!

**Review by Holli-Git — CHANGES_REQUESTED** [review]

File: data/json/items/gun/robofac_gun.json (hunk 4 above)
Comment (Holli-Git) [reply data/json/items/gun/robofac_gun.json:461]: Well then test it

**Comment by Holli-Git** [comment]
I'm uncertain that this would still be warranted. The reasons you have/had for this aren't nearly as concrete as I'd like, and not to mention you still haven't fixed the issue of the stats of the attachment. 

Overall, I'm not confident that this is needed or would make sense for the hub to have. 300 BLK is quite a rare caliber, and while it fits a niche, it's a niche that can very easily be covered by other calibers. The PR was also based on a wrong assumption about how common it is in military settings, and without that basis I'm uncertain why we need this conversion.

**Comment by kevingranade** [comment]
Why is robofac even considering .300 to be an option? It's like 10x rarer than 7.62 or 5.56. If you want a more powerful round than those, even .30-06 is more sensible. 

They're in survival/scavenging mode and they know it,  they're not going to make variabts for rare ammo types, they're going to focus on things that are readily available.

**Pull request closed without merging**